Source/ACE/Entity/Player.cs

```diff
@@ -1763,33 +1794,106 @@ public void HandleActionWieldItem(Container container, uint itemId, uint locatio
                             HandlePickupItem(container, itemGuid, location, PropertyInstanceId.Wielder);
                             return;
                         }
+
                         // Ok do you have this - if not bail
-                        if (GetInventoryItem(itemGuid) == null) return;
+                        var item = GetInventoryItem(itemGuid);
+                        if (item == null)
+                            return;
+                        if ((EquipMask)location > EquipMask.RingLeft)
```

Comment:
These nested conditionals are getting messy.  We should think if there is a more elegant solution to this problem, or they should be commented very well.